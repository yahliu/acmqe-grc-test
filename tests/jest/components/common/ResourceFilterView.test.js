/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import ResourceFilterView from '../../../../src-web/components/common/ResourceFilterView'
import renderer from 'react-test-renderer'

describe('ResourceFilterView component', () => {
  it('renders as expected', () => {
    const activeFilters = {
      'standards': [
        'HIPAA'
      ]
    }
    const availableFilters = {
      'standards': {
        'name': 'Standards',
        'availableSet': [
          'NIST',
          'FISMA',
          'PCI',
          'HIPAA'
        ]
      },
      'categories': {
        'name': 'Categories',
        'availableSet': [
          'System And Information Integrity',
          'System And Communications Protections'
        ]
      },
      'controls': {
        'name': 'Controls',
        'availableSet': [
          'VA',
          'Secret Encryption',
          'Cert Manager',
          'Mutation Advisor'
        ]
      },
      'type': {
        'name': 'Type',
        'availableSet': [
          'Enforce',
          'Inform only'
        ]
      },
      'severity': {
        'availableSet': []
      }
    }
    const fn = jest.fn()
    const component = renderer.create(
      <ResourceFilterView
        updateFilters={fn}
        onClose={fn}
        activeFilters={activeFilters}
        availableFilters={availableFilters}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('ResourceFilterView component no filter', () => {
  it('renders as expected', () => {
    const fn = jest.fn()
    const component = renderer.create(
      <ResourceFilterView
        updateFilters={fn}
        onClose={fn}
        activeFilters={{}}
        availableFilters={{}}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
