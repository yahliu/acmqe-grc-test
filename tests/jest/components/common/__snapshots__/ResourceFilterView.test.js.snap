// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceFilterView component no filter renders as expected 1`] = `
<div
  className="resource-filter-view"
  style={
    Object {
      "height": 771,
    }
  }
>
  <h3
    className="filterHeader"
  >
    <svg
      className="titleIcon"
    >
      <use
        href="#diagramIcons_filter"
      />
    </svg>
    <div
      className="titleText"
    >
      Filter by
    </div>
    <svg
      alt="Close filter view"
      aria-label="Close filter view"
      className="closeIcon"
      fillRule="evenodd"
      height="10"
      name="icon--close"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
      viewBox="0 0 10 10"
      width="10"
    >
      <title>
        Close filter view
      </title>
      <path
        d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
      />
    </svg>
  </h3>
  <div
    className="filter-sections-container"
    style={
      Object {
        "height": 688,
        "overflow": "hidden",
        "position": "relative",
        "width": 260,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "bottom": 0,
          "left": 0,
          "marginBottom": -17,
          "marginRight": 0,
          "overflow": "scroll",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "height": 6,
          "left": 2,
          "position": "absolute",
          "right": 2,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "height": "100%",
            "position": "relative",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "position": "absolute",
          "right": 2,
          "top": 2,
          "width": 6,
        }
      }
    >
      <div
        className="filter-sections-scrollbar"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "position": "relative",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ResourceFilterView component renders as expected 1`] = `
<div
  className="resource-filter-view"
  style={
    Object {
      "height": 771,
    }
  }
>
  <h3
    className="filterHeader"
  >
    <svg
      className="titleIcon"
    >
      <use
        href="#diagramIcons_filter"
      />
    </svg>
    <div
      className="titleText"
    >
      Filter by
    </div>
    <svg
      alt="Close filter view"
      aria-label="Close filter view"
      className="closeIcon"
      fillRule="evenodd"
      height="10"
      name="icon--close"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
      viewBox="0 0 10 10"
      width="10"
    >
      <title>
        Close filter view
      </title>
      <path
        d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
      />
    </svg>
  </h3>
  <div
    className="filter-sections-container"
    style={
      Object {
        "height": 688,
        "overflow": "hidden",
        "position": "relative",
        "width": 260,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "bottom": 0,
          "left": 0,
          "marginBottom": -17,
          "marginRight": 0,
          "overflow": "scroll",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        >
          Standards
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="standardsFISMA"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="standardsFISMA"
            title={null}
          >
            <span
              className=""
            >
              FISMA
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="standardsHIPAA"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="standardsHIPAA"
            title={null}
          >
            <span
              className=""
            >
              HIPAA
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="standardsNIST"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="standardsNIST"
            title={null}
          >
            <span
              className=""
            >
              NIST
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="standardsPCI"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="standardsPCI"
            title={null}
          >
            <span
              className=""
            >
              PCI
            </span>
          </label>
        </div>
      </div>
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        >
          Categories
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="categoriesSystem And Communications Protections"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="categoriesSystem And Communications Protections"
            title={null}
          >
            <span
              className=""
            >
              System And Communications Protections
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="categoriesSystem And Information Integrity"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="categoriesSystem And Information Integrity"
            title={null}
          >
            <span
              className=""
            >
              System And Information Integrity
            </span>
          </label>
        </div>
      </div>
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        >
          Controls
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="controlsCert Manager"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="controlsCert Manager"
            title={null}
          >
            <span
              className=""
            >
              Cert Manager
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="controlsMutation Advisor"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="controlsMutation Advisor"
            title={null}
          >
            <span
              className=""
            >
              Mutation Advisor
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="controlsSecret Encryption"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="controlsSecret Encryption"
            title={null}
          >
            <span
              className=""
            >
              Secret Encryption
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="controlsVA"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="controlsVA"
            title={null}
          >
            <span
              className=""
            >
              VA
            </span>
          </label>
        </div>
      </div>
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        >
          Type
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="typeEnforce"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="typeEnforce"
            title={null}
          >
            <span
              className=""
            >
              Enforce
            </span>
          </label>
        </div>
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="typeInform only"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label filter-section-checkbox"
            htmlFor="typeInform only"
            title={null}
          >
            <span
              className=""
            >
              Inform only
            </span>
          </label>
        </div>
      </div>
      <div
        className="filter-section"
      >
        <div
          className="filter-section-title"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "height": 6,
          "left": 2,
          "position": "absolute",
          "right": 2,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "height": "100%",
            "position": "relative",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "position": "absolute",
          "right": 2,
          "top": 2,
          "width": 6,
        }
      }
    >
      <div
        className="filter-sections-scrollbar"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "position": "relative",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
